### ISSUE TITLE
warning: ignoring return value of ‘asprintf’, declared with attribute warn_unused_result

### ISSUE THREAD

**Issue by honggyukim** [namhyung/uftrace#1453]
There are some warnings when building `unittest` as follows.
```
$ make unittest
      ...
/home/honggyu/work/uftrace/utils/data-file.c: In function ‘create_data’:
/home/honggyu/work/uftrace/utils/data-file.c:666:2: warning: ignoring return value of ‘asprintf’, declared with attribute warn_unused_result [-Wunused-result]
  666 |  asprintf(&filename, "%s/sid-%.*s.map",
      |  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  667 |    opts->dirname, SESSION_ID_LEN, TEST_SESSION_ID);
      |    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/honggyu/work/uftrace/utils/data-file.c:672:2: warning: ignoring return value of ‘asprintf’, declared with attribute warn_unused_result [-Wunused-result]
  672 |  asprintf(&filename, "%s/%d.dat", opts->dirname, task_msg.pid);
      |  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      ...
```
We better fix those warnings.

**Comment by SulRash** [comment]
Hello,

This issue seems easy to fix on the surface and I just wanted to ask if the bytes returned from asprintf are used? If not, I think I can push a fix pretty quickly.

**Comment by honggyukim** [comment]
Hi @SulRash, sorry, the correct fix is to check the return value rather than ignoring it. @kangtegong already made the fix in #1489 and it's merged. So I will close this issue and your PR together.
